src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -1092,15 +1126,21 @@ PropertyPath Visitor::visitTypesafe(Parser::PathEltOrInverseContext* ctx) {
 }
 
 // ____________________________________________________________________________________
-PropertyPath Visitor::visitTypesafe(Parser::PathPrimaryContext* ctx) {
+void Visitor::visit(Parser::PathModContext*) {
+  // PathMod should be handled by upper clauses. It should not be visited.
+  AD_FAIL();
+}
+
+// ____________________________________________________________________________________
+PropertyPath Visitor::visit(Parser::PathPrimaryContext* ctx) {
   // TODO<qup42, joka921> Is there a way to make this visitAlternative in the
   // presence of the a case?
```

Comment:
I think this TODO can be updated:
It is now something like "TODO: implement a strong `Iri` type, s.t. the `ctx->iri()` case can become a simple `return visit(...)`. Then the three cases which are not the special `a` case can be merged into a `visitAlternative`."